src/infinity-menu.js

```diff
@@ -261,6 +261,7 @@ export default class InfinityMenu extends React.Component {
 
 		/*header component*/
 		const headerProps = {
+			headerMetadata: this.props.headerMetadata,
```

Comment:
i still think we'd be better off using object rest spread here, it makes this much more flexible for user defined components external to this component
```js
const Header = ({title}) => <div>{title}</div>;
```
is much nicer to me then
const Header = ({headerMetadata}) => <div>{headerMetadata.title}</div>;

this also makes it a lot easier for a user to use predefined components, or override props that are being passed into it be default if we execute. the spread at the end of the object.